sites/cheerpj/src/content/docs/11-guides/basic-server-setup.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Basic Server Setup
+description: How do serve your own Java application with CheerpJ
+---
+
+## Overview
+
+To run your Java applications in the browser, you will need to self host the application files. This guide will cover some simple HTTP servers that can be used for basic testing and will discuss NGINX for more advanced setups, for example when self hosting CheerpJ.
+
+> [!note] Important
+> The content of this guide is specifically made for testing setups and should not be used for production environments. For production environments, it is essential to utilize HTTPS, since it is more secure than plain HTTP.
+
+## Static "one-line" servers
+
+One-line HTTP servers are simple tools that can be run directly from the command line with a short command. They are simple to set up and are useful for quickly serving static files or testing web applications locally.
+
+_One-Line_ servers that have been tested and work with CheerpJ:
+
+- `npx serve`
+- `npx http-server`
+
+## CheerpJ HTTP server requirements
+
+### Range requests
+
+CheerpJ makes use of [Range Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Range_requests), to download resources in chunks, for example JAR files.
+Thus, the HTTP server hosting the Java application files must support "Range" headers. Some HTTP servers may need additional configuration to enable this feature, while others have native support, and some are not compatible at all. For instance, Python's `http-server` module, commonly used for quick testing, lacks support for range headers and therefore won't work with CheerpJ.
+
+If your HTTP server does not support `Range Requests`, you will see an error message like this in the browser console:
+
+> [!warning] Network error for http://127.0.0.1:8084/exampleApp.jar: HTTP server does not support the 'Range' header. CheerpJ cannot run.
+
+NPM server tools such as `npx serve` and `npx http-server` typically have built-in support for range requests, making them suitable for basic testing without the need for additional configurations. The same applies to NGINX, which will be discussed in more detail later in this guide.
+
+### JavaScript library files
+
+CheerpJ allows you to implement native libraries in JavaScript, more details on this can be found in our [Native Libraries Guide](/docs/guides/implementing-native-librariest).
+
+Depending on the size of the JavaScript library file, you may encounter an error with certain browsers, like Firefox:
+
+> [!warning] Network error for http://localhost:8080/library.js: HTTP server returned compressed partial data. That should not happen. CheerpJ cannot run
+
+This error occurs because plain JavaScript files are compressed automatically in Firefox, since it provides the `Accept-Encoding: gzip` header for range requests, which is illegal. It is unclear whether the response contains partial compressed data, or compressed partial data.
```

Comment:
```suggestion
This error occurs because plain JavaScript files are compressed automatically in Firefox, since it provides the `Accept-Encoding: gzip` header for range requests. This is not well defined, since it is unclear whether the response contains partial compressed data, or compressed partial data.
```